allensdk/brain_observatory/behavior/eye_tracking_processing.py

```diff
@@ -0,0 +1,164 @@
+
+
+def load_eye_tracking_hdf(eye_tracking_file: Path) -> pd.DataFrame:
+    """Load a DeepLabCut hdf5 file containing eye tracking data into a
+    dataframe.
+
+    Note: The eye tracking hdf5 file contains 3 separate dataframes. One for
+    corneal reflection (cr), eye, and pupil ellipse fits. This function
+    loads and returns this data as a single dataframe.
+
+    Parameters
+    ----------
+    eye_tracking_file : Path
+        Path to an hdf5 file produced by the DeepLabCut eye tracking pipeline.
+        The hdf5 file will contain the following keys: "cr", "eye", "pupil".
+        Each key has an associated dataframe with the following
+        columns: "center_x", "center_y", "height", "width", "phi".
+
+    Returns
+    -------
+    pd.DataFrame
+        A dataframe containing combined corneal reflection (cr), eyelid (eye),
+        and pupil data. Column names for each field will be renamed by
+        prepending the field name. (e.g. center_x -> eye_center_x)
+    """
+    eye_tracking_fields = ["cr", "eye", "pupil"]
+
+    eye_tracking_dfs = []
+    for field_name in eye_tracking_fields:
+        field_data = pd.read_hdf(eye_tracking_file, key=field_name)
+        new_col_name_map = {col_name: f"{field_name}_{col_name}"
+                            for col_name in field_data.columns}
+        field_data.rename(new_col_name_map, axis=1, inplace=True)
+        eye_tracking_dfs.append(field_data)
+
+    eye_tracking_data = pd.concat(eye_tracking_dfs, axis=1)
+    eye_tracking_data.index.name = 'frame'
+
+    # Values in the hdf5 may be complex (likely an artifact of the ellipse
+    # fitting process). Take only the real component.
+    eye_tracking_data = eye_tracking_data.apply(lambda x: np.real(x.to_numpy()))
+
+    return eye_tracking_data.astype(float)
+
+
+def determine_outliers(eye_tracking_df: pd.DataFrame,
+                       z_threshold: float) -> np.ndarray:
+    """Given an eye tracking dataframe and some z-score threshold return
+    a set of frames indices whose column values exceed the z-score threshold.
+
+    Parameters
+    ----------
+    eye_tracking_df : pd.DataFrame
+        An eye tracking dataframe produced by load_eye_tracking_hdf().
+    z_threshold : float
+        z-score values higher than the z_threshold will be considered outliers.
+
+    Returns
+    -------
+    np.ndarray
+        A numpy boolean array whose length == len(eye_tracking_df.index).
+        True denotes that a row in the eye_tracking_df contains an outlier.
+    """
+
+    # Dataframe must have NANs filled to prevent warning when performing
+    # comparisons against the z_threshold
+    nan_filled_df = eye_tracking_df.fillna(eye_tracking_df.mean())
+
+    outliers = (np.abs(stats.zscore(nan_filled_df, axis=1)) > z_threshold)
+    return pd.Series(outliers.any(axis=1))
+
+
+def compute_pupil_area(df_row: pd.Series) -> float:
+    # Calculate the area of the pupil as a circle using the max of the
+    # height/width as radius
```

Comment:
I might suggest fleshing out this comment a bit. Add:
# This assumes that the pupil is a perfect circle and any eccentricity is a result of the angle at which the pupil is being viewed